addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/ScriptApi.java

```diff
@@ -0,0 +1,510 @@
+            }
+            extension.removeScript(script);
+            return ApiResponseElement.OK;
+
+        } else if (ACTION_RUN_STANDALONE.equals(name)) {
+            ScriptWrapper script = extension.getScript(params.getString(ACTION_PARAM_SCRIPT_NAME));
+            if (script == null) {
+                throw new ApiException(ApiException.Type.DOES_NOT_EXIST, ACTION_PARAM_SCRIPT_NAME);
+            }
+            if (!script.getType().getName().equals(ExtensionScript.TYPE_STANDALONE)) {
+                throw new ApiException(
+                        ApiException.Type.ILLEGAL_PARAMETER,
+                        "Parameter "
+                                + ACTION_PARAM_SCRIPT_NAME
+                                + " does not match a "
+                                + ExtensionScript.TYPE_STANDALONE
+                                + " script.");
+            }
+            try {
+                extension.invokeScript(script);
+            } catch (Exception e) {
+                throw new ApiException(ApiException.Type.INTERNAL_ERROR, e);
+            }
+            return ApiResponseElement.OK;
+
+        } else if (ACTION_CLEAR_GLOBAL_VAR.equals(name)) {
+            ScriptVars.setGlobalVar(params.getString(PARAM_VAR_KEY), null);
+            return ApiResponseElement.OK;
+        } else if (ACTION_CLEAR_GLOBAL_CUSTOM_VAR.equals(name)) {
+            ScriptVars.setGlobalCustomVar(params.getString(PARAM_VAR_KEY), null);
+            return ApiResponseElement.OK;
+        } else if (ACTION_CLEAR_GLOBAL_VARS.equals(name)) {
+            ScriptVars.clearGlobalVars();
+            return ApiResponseElement.OK;
+        } else if (ACTION_CLEAR_SCRIPT_VAR.equals(name)) {
+            ScriptVars.setScriptVar(
+                    getAndValidateScriptName(params), params.getString(PARAM_VAR_KEY), null);
+            return ApiResponseElement.OK;
+        } else if (ACTION_CLEAR_SCRIPT_CUSTOM_VAR.equals(name)) {
+            ScriptVars.setScriptCustomVar(
+                    getAndValidateScriptName(params), params.getString(PARAM_VAR_KEY), null);
+            return ApiResponseElement.OK;
+        } else if (ACTION_CLEAR_SCRIPT_VARS.equals(name)) {
+            ScriptVars.clearScriptVars(getAndValidateScriptName(params));
+            return ApiResponseElement.OK;
+        } else if (ACTION_SET_GLOBAL_VAR.equals(name)) {
+            ScriptVars.setGlobalVar(
+                    params.getString(PARAM_VAR_KEY), params.getString(PARAM_VAR_VALUE));
+            return ApiResponseElement.OK;
+        } else if (ACTION_SET_SCRIPT_VAR.equals(name)) {
+            ScriptVars.setScriptVar(
+                    getAndValidateScriptName(params),
+                    params.getString(PARAM_VAR_KEY),
+                    params.getString(PARAM_VAR_VALUE));
+            return ApiResponseElement.OK;
+        } else {
+            throw new ApiException(ApiException.Type.BAD_ACTION);
```

Comment:
This line was changed compared to core code, previously `BAD_VIEW`.

Reply:
😄 good catch

Reply:
Tests caught it :)